### DISCUSSION TITLE
Trying to suppress deprecation warnings with PHP 8.1 and Drupal 9

### DISCUSSION THREAD

**Discussion by synterra** [phpro/grumphp#1073] [Q&A]
I've found myself in a bit of a conundrum. I'm using version 1.5.1 of grumphp on a Drupal 9 site, that I'm working on upgrading to Drupal 10. To do that, I have to first upgrade from PHP 8.0 to 8.1. When I do that, every time I commit code and grumphp kicks in, I get piles of deprecation warnings.

The obvious solution is to upgrade to grumphp 1.8+ which supports PHP 8.1. However, that also requires Symfony 5+ libraries, and Drupal 9 is using Symfony 4. Ironically, Drupal 10 uses Symfony 5, so things should work fine once everything is fully updated. But I'm a ways from being ready to make that upgrade.

Since I can't upgrade, I've been trying to suppress the warnings, and I've had some success by modifying `EXEC_GRUMPHP_COMMAND` in grumphp.yml, but even this only stops some of them:
```
grumphp:
  git_hook_variables:
    EXEC_GRUMPHP_COMMAND: [
      'ddev', 'exec', 'php',
      '-d error_reporting="E_NONE"',
      '-d ignore_repeated_errors="Off"',
      '-d display_errors="Off"',
      '-d display_startup_errors="Off"',
      '-d log_errors="Off"'
    ]
```

Anyone have any other ideas on how to suppress the notices until I can fully upgrade?

**Comment by veewee** [comment]
You could try to use a never version of grumphp through the dependencyless `grumphp-shim` package.
That way it uses newer symfony dependencies internally.
